pkg/git/gitclient/gitclient.go

```diff
@@ -268,11 +269,26 @@ func (g *GitClient) Branch(name string) error {
 
 func (g *GitClient) ValidateRemoteExists(ctx context.Context) error {
 	logger.V(3).Info("Validating git setup", "repoUrl", g.RepoUrl)
-	remote := g.Client.NewRemote(g.RepoUrl, gogit.DefaultRemoteName)
-	// Check if we are able to make a connection to the remote by attempting to list refs
-	_, err := g.Client.ListWithContext(ctx, remote, g.Auth)
+
+	isCodecommit, codeCommitUser, err := git.IsCodeCommitURL(g.RepoUrl)
 	if err != nil {
-		return fmt.Errorf("connecting with remote %v for repository: %v", gogit.DefaultRemoteName, err)
+		return err
+	}
+	if isCodecommit {
+		logger.V(6).Info("Validating codecommit url")
+		if codeCommitUser == nil || codeCommitUser.Username() == "" {
+			return fmt.Errorf("invalid AWS CodeCommit url: ssh key id should be specified in the url")
+		}
+		if codeCommitUser.Username() == constants.DefaultSSHAuthUser {
```

Comment:
Is it possible for the `codeCommitUser.Username()` to be "git"?

Reply:
The ssh key id is generated by aws as user unique identifier. https://docs.aws.amazon.com/codecommit/latest/userguide/setting-up-ssh-unixes.html#setting-up-ssh-unixes-keys
Even tho the one of the identifier happens to be `git`, as long as it pairs with local ssh private key, ssh call should still work.